pandas/_libs/groupby_helper.pxi.in

```diff
@@ -53,19 +41,15 @@ def group_last_{{name}}({{c_type}}[:, :] out,
         raise AssertionError("len(index) != len(labels)")
 
     nobs = np.zeros((<object>out).shape, dtype=np.int64)
-    {{if name == 'object'}}
-    resx = np.empty((<object>out).shape, dtype=object)
-    {{else}}
-    resx = np.empty_like(out)
-    {{endif}}
+    if rank_t is object:
+        resx = np.empty((<object>out).shape, dtype=object)
+    else:
+        resx = np.empty_like(out)
 
     N, K = (<object>values).shape
 
-    {{if name == "object"}}
-    if True:  # make templating happy
-    {{else}}
-    with nogil:
-    {{endif}}
+    if rank_t is object:
```

Comment:
I love the fused types here but can we just wait for Cython 0.30 to get released? I might be overlooking the benefit in the interim but feel like there is only risk to changing this conditional nogil only to revert back in the future

Reply:
> the benefit in the interim

In the ongoing catch-less process in core.groupby I'm finding that the lack of uint64_t versions of some of these functions is a blocker (presumably with perf impact).  I find it easier to implement the uint64_t version with fused types than without, so figured get this out of the way.